lightnode.go

```diff
@@ -54,5 +68,20 @@ func (node *Lightnode) Run(done <-chan struct{}) {
 			node.logger.Errorf("failed to serve: %v", err)
 		}
 	}()
-	node.resolver.Run(done)
+
+	go node.resolver.Run(done)
+	node.resolver.IO().InputWriter() <- p2p.Tick{}
+
+	ticker := time.NewTicker(5 * time.Minute)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-done:
+			return
+		case <-ticker.C:
+			node.logger.Debug("updating darknode multi addresses")
+			node.resolver.IO().InputWriter() <- p2p.Tick{}
```

Comment:
Once the task is running, you can no longer interact with `node.resolver.IO().InputWriter()` safely. It might work in this particular instance, but generally the external IO pattern needs to be used. I would recommend that an explicit timer task is created.